src/test/java/com/alvarium/exampleapp/observers/MutationObserverTest.java

```diff
@@ -0,0 +1,78 @@
+package com.alvarium.exampleapp.observers;
+
+import com.alvarium.Sdk;
+import com.alvarium.SdkInfo;
+import com.alvarium.exampleapp.SampleData;
+import com.alvarium.sign.KeyInfo;
+import com.alvarium.utils.PropertyBag;
+
+import org.junit.Test;
+
+import config.Reader;
+import config.ReaderFactory;
+import config.ReaderType;
+import io.reactivex.rxjava3.core.Observer;
+
+public class MutationObserverTest {
+  @Test
+  public void mutationObserverShouldMutateAndPublish() throws Exception{
+    
+    Channel<SampleData> srcChannel = new Channel<SampleData>();
+    Channel<SampleData> destinationChannel =new Channel<SampleData>();
+    
+    final Sdk sdk = new Sdk() {
+      @Override
+      public void create(byte[] data) {
+        System.out.println("Creating data");
+        System.out.println(new String(data));
+      }
+
+      @Override
+      public void create(PropertyBag ctx, byte[] data) {
+        System.out.println(new String(data));
+      }
+
+      @Override
+      public void mutate(byte[] oldData, byte[] newData) {
+        System.out.println(new String(newData));
+      }
+
+      @Override
+      public void mutate(PropertyBag ctx, byte[] oldData, byte[] newData) {
+        System.out.println(new String(newData));
+      }
+
+      @Override
+      public void transit(byte[] data) {
+        System.out.println(new String(data));
+      }
+
+      @Override
+      public void transit(PropertyBag ctx, byte[] data) {
+        System.out.println(new String(data));
+      }
+
+      @Override
+      public void close() {
+
+      }
+    };
+    ReaderFactory readerFactory = new ReaderFactory();
+    Reader reader = readerFactory.getReader(ReaderType.JSON);
+
+    SdkInfo sdkInfo = reader.read("./src/main/resources/config.json");
+    final KeyInfo keyInfo = new KeyInfo(
```

Comment:
same as the previous comment. can be replaced by `final KeyInfo keyinfo = sdkInfo.getSignature().getPrivateKey()`